tests/templates/kuttl/smoke/60-install-hive.yaml.j2

```diff
@@ -26,9 +26,24 @@ spec:
     config:
       logging:
         enableVectorAgent: {{ lookup('env', 'VECTOR_AGGREGATOR') | length > 0 }}
+    envOverrides:
+      COMMON_VAR: role-value # overridden by role group below
+      ROLE_VAR: role-value   # only defined here at role level
+    configOverrides:
```

Comment:
I would suggest adding some test cases covering the whole override cycle. For example adding
```
configOverrides:
  hive-site.xml:
    hive.metastore.metrics.enabled: "false"
```
to the HiveCluster and the test
```
- script: |
      kubectl -n $NAMESPACE get cm hive-metastore-default -o yaml | yq -e '.data."hive-site.xml"' | grep "<name>hive.metastore.metrics.enabled</name>" -A 1 | grep "<value>false</value>"
```
to the assert, would reproduce the error of configOverrides not working properly.

Reply:
I don't understand. This is exactly what is tested with `hive-env.sh`.

Reply:
I'm a bit meh on the shadowing thing. It feels very naive. One can go around it any time by using a different var.
I would prefer to make `merge_from` composable.

Updated the op-rs version here: https://github.com/stackabletech/hive-operator/pull/507/commits/62e0b97b3f2fd48aa229bef456e1ce97e401e5a8

Reply:
> I'm a bit meh on the shadowing thing. It feels very naive. One can go around it any time by using a different var. I would prefer to make `merge_from` composable.

Just a note: For now we decided to leave that out and address the merging composability during later refactorings.

Reply:
Updated test: https://github.com/stackabletech/hive-operator/pull/507/commits/cf3a537f1918626058557aba70ef2d649cf85432

:green_circle: Test

```
--- PASS: kuttl (136.86s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/smoke_postgres-12.5.6_hive-3.1.3_openshift-false_s3-use-tls-false (136.20s)
PASS
```